go/tools/releasing/cmd/prerelease.go

```diff
@@ -0,0 +1,309 @@
+		existingTags[strings.TrimSpace(tag)] = true
+	}
+
+	for _, newFullTag := range modFullTags {
+		if existingTags[newFullTag] {
+			return fmt.Errorf("git tag already exists for %v", newFullTag)
+		}
+	}
+
+	return nil
+}
+
+// verifyWorkingTreeClean checks if the working tree is clean (i.e. running 'git diff --exit-code' gives exit code 0).
+// If the working tree is not clean, the git diff output is printed, and an error is returned.
+func (p prerelease) verifyWorkingTreeClean() error {
+	cmd := exec.Command("git", "diff", "--exit-code")
+	if output, err := cmd.CombinedOutput(); err != nil {
+		return fmt.Errorf("working tree is not clean, can't proceed with the release process:\n\n%v",
+			string(output),
+		)
+	}
+
+	return nil
+}
+
+func (p prerelease) createPrereleaseBranch(fromExistingBranch string) error {
+	branchNameElements := []string{"pre_release", p.modSetName, p.newVersion}
+	branchName := strings.Join(branchNameElements, "_")
+	fmt.Printf("git checkout -b %v %v\n", branchName, fromExistingBranch)
+	cmd := exec.Command("git", "checkout", "-b", branchName, fromExistingBranch)
+
+	if output, err := cmd.CombinedOutput(); err != nil {
+		return fmt.Errorf("could not create new branch %v: %v (%v)", branchName, string(output), err)
+	}
+
+	return nil
+}
+
+// TODO: updateVersionGo may be implemented to update any hard-coded values within version.go files as needed.
+func (p prerelease) updateVersionGo() error {
+	return nil
+}
+
+// runMakeLint runs 'make lint' to automatically update go.sum files.
+func (p prerelease) runMakeLint() error {
+	fmt.Println("Updating go.sum with 'make lint'...")
+
+	cmd := exec.Command("make", "lint")
+	if output, err := cmd.CombinedOutput(); err != nil {
+		return fmt.Errorf("'make lint' failed: %v (%v)", string(output), err)
+	}
+
+	return nil
+}
+
+func (p prerelease) commitChanges(skipMake bool) error {
+	commitMessage := "Prepare for releasing " + p.newVersion
+
+	// add changes to git
+	cmd := exec.Command("git", "add", ".")
+	if output, err := cmd.CombinedOutput(); err != nil {
+		return fmt.Errorf("'git add .' failed: %v (%v)", string(output), err)
+	}
+
+	// make ci
+	if skipMake {
+		fmt.Println("Skipping 'make ci'...")
+	} else {
+		fmt.Println("Running 'make ci'...")
+		cmd = exec.Command("make", "ci")
+		if output, err := cmd.CombinedOutput(); err != nil {
+			return fmt.Errorf("'make ci' failed: %v (%v)", string(output), err)
+		}
+	}
+
+	// commit changes to git
+	fmt.Printf("Commit changes to git with message '%v'...\n", commitMessage)
+	cmd = exec.Command("git", "commit", "-m", commitMessage)
+	if output, err := cmd.CombinedOutput(); err != nil {
+		return fmt.Errorf("git commit failed: %v (%v)", string(output), err)
+	}
+
+	cmd = exec.Command("git", "log", `--pretty=format:"%h"`, "-1")
+	output, err := cmd.Output()
+	if err != nil {
+		fmt.Println("WARNING: could not automatically get last commit hash.")
+	}
+
+	fmt.Println("Commit successful. Hash of commit:")
+	os.Stdout.Write(output)
```

Comment:
```suggestion
	fmt.Printf("Commit successful. Hash of commit: %s\n", output)
```

This should work, no?  Slightly clearer IMO.